drivers/jtag/jtag-aspeed.c

```diff
@@ -667,15 +672,27 @@ static int aspeed_jtag_shctrl_tms_mask(enum jtag_tapstate from,
 				       u32 start_shift, u32 end_shift,
 				       u32 *tms_mask)
 {
+	// _tms_cycle_lookup[16][4].count
 	u32 pre_tms = start_shift ? _tms_cycle_lookup[from][to].count : 0;
 	u32 post_tms = end_shift ? _tms_cycle_lookup[there][endstate].count : 0;
 	u32 tms_value = start_shift ? _tms_cycle_lookup[from][to].tmsbits : 0;
 
+	pr_debug("start_shift is set to %u", start_shift);
+	pr_debug("end_shift   is set to %u", end_shift);
+	pr_debug("pre_tms     is set to %u", pre_tms);
+	pr_debug("post_tms    is set to %u", post_tms);
+	pr_debug("tms_value   is set to %u", tms_value);
+	pr_debug("tms_cycle_lookup        from-to count is set to %u", _tms_cycle_lookup[from][to].count);
+	pr_debug("tms_cycle_lookup there-endstate count is set to %u", _tms_cycle_lookup[there][endstate].count);
+	pr_debug("tms_cycle_lookup      from-to tmsbits is set to %u", _tms_cycle_lookup[from][to].tmsbits);
+	pr_debug("tms_mask pointer is %p and value %u", (void*)tms_mask, *tms_mask);
+
 	tms_value |= end_shift ? _tms_cycle_lookup[there][endstate].tmsbits
 					 << pre_tms :
 				 0;
 	if (pre_tms > GENMASK(2, 0) || post_tms > GENMASK(2, 0)) {
-		pr_err("pre/port tms count is greater than hw limit");
+		pr_err("pre (0x%x)/post (0x%x) tms count is greater than hw limit",
```

Comment:
dev_err

Reply:
I think that dev_dbg an dev_err are not available within the scope of this function.

Reply:
removed it for now